source/common/http/filter_chain_helper.h

```diff
@@ -13,48 +13,146 @@
 namespace Envoy {
 namespace Http {
 
-using FilterConfigProviderManager =
+using DownstreamFilterConfigProviderManager =
     Filter::FilterConfigProviderManager<Filter::NamedHttpFilterFactoryCb,
                                         Server::Configuration::FactoryContext>;
+using UpstreamFilterConfigProviderManager =
+    Filter::FilterConfigProviderManager<Filter::NamedHttpFilterFactoryCb,
+                                        Server::Configuration::ServerFactoryContext>;
```

Comment:
So this means the upstream ones get passed two ServerFactoryContext - is this just temporary until we get an upstream specific factory context?

Reply:
Correct.  90% sure we're going to want an UpstreamFactoryContext too, so that we can hand it the cluster scope and cluster init manager to not leak stats and correctly wait on initialization when we reload clusters.